app/src/main/java/app/mywatchlist/screens/MainScreen.kt

```diff
@@ -0,0 +1,69 @@
+package app.mywatchlist
+
+
+import androidx.compose.foundation.layout.RowScope
+import androidx.compose.material.BottomNavigation
+import androidx.compose.material.BottomNavigationItem
+import androidx.compose.material.Scaffold
+import androidx.compose.material.Text
+import androidx.compose.material3.Icon
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.navigation.NavDestination
+import androidx.navigation.NavDestination.Companion.hierarchy
+import androidx.navigation.NavGraph.Companion.findStartDestination
+import androidx.navigation.NavHostController
+import androidx.navigation.compose.currentBackStackEntryAsState
+import androidx.navigation.compose.rememberNavController
+
+@Composable
+fun MainScreen(){
+    val navController = rememberNavController()
+    Scaffold(
+        bottomBar = { BottomBar(navController = navController) }
+    ) {
+        SetupNavGraph(navController = navController)
+    }
+}
+
+@Composable
+fun BottomBar(navController: NavHostController){
+    val screens = listOf(
+        Screen.Home,
+        Screen.MyWatchlist,
+        Screen.Profile
+    )
+    val navBackStackEntry by navController.currentBackStackEntryAsState()
+    val currentDestination = navBackStackEntry?.destination
+
+    BottomNavigation {
+        screens.forEach { screen ->
+            AddItem(screen = screen, currentDestination = currentDestination, navController = navController)
```

Comment:
JFYI: Normally you only need the named arguments if you don't supply them in the right order. If `AddItem` expects first the screen, then the current desination and than the navController, you could also just write
```suggestion
            AddItem(screen, currentDestination, navController)
```
I think this is the case here, also applies for some other cases.